azurerm/resource_arm_iothub_fallback_route.go

```diff
@@ -0,0 +1,190 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/iothub/mgmt/2018-12-01-preview/devices"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmIotHubFallbackRoute() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmIotHubFallbackRouteCreateUpdate,
+		Read:   resourceArmIotHubFallbackRouteRead,
+		Update: resourceArmIotHubFallbackRouteCreateUpdate,
+		Delete: resourceArmIotHubFallbackRouteDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"iothub_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.IoTHubName,
+			},
+
+			"condition": {
+				// The condition is a string value representing device-to-cloud message routes query expression
+				// https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-query-language#device-to-cloud-message-routes-query-expressions
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "true",
+			},
+
+			"endpoint_names": {
+				Type:     schema.TypeList,
+				Required: true,
+				// Currently only one endpoint is allowed. With that comment from Microsoft, we'll leave this open to enhancement when they add multiple endpoint support.
+				MaxItems: 1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
```

Comment:
We should use `validate.IoTHubEndpointName` here?

Reply:
Good call